### PR TITLE
Docs update: Adds supertokens to 'Bring Your Own Database' section as well + improvements to code structure in example

### PR DESCRIPTION
**Pull request by rishabhpoddar**
- Since SuperTokens is a self hosted solution, it can be added to 'Bring Your Own Database' as well (in https://nextjs.org/docs/authentication#supertokens)
- Improvements to code structure in the with-supertokens example

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
Hey! Anything I can do to help review this PR please? Thanks.

File: docs/authentication.md [hunk 1]
```diff
@@ -127,6 +127,8 @@ Both of these libraries support either authentication pattern. If you're interes
 - [with-passport](https://github.com/vercel/next.js/tree/canary/examples/with-passport)
 - [with-passport-and-next-connect](https://github.com/vercel/next.js/tree/canary/examples/with-passport-and-next-connect)
 
+- [SuperTokens](/docs/authentication#supertokens) (more info below) also allows you to store data within your own database.
+
```
Comment (leerob) [inline docs/authentication.md:131]:
```suggestion
- [SuperTokens](/docs/authentication#supertokens) (more info below) also allows you to store data within your own database.

```

This is already below 👍

Comment (leerob) [inline docs/authentication.md:131]:
```suggestion
```

This is already below :+1:

**Review by leerob — COMMENTED** [review]
LGTM other than that!

**Comment by rishabhpoddar** [comment]
@leerob accepted your change. Thanks.

**Review by leerob — APPROVED** [review]

**Pull request merged**